public/app/features/explore/utils/decorators.ts

```diff
@@ -86,7 +91,9 @@ export const decorateWithGraphResult = (data: ExplorePanelData): ExplorePanelDat
  * multiple results and so this should be used with mergeMap or similar to unbox the internal observable.
  */
 export const decorateWithTableResult = (data: ExplorePanelData): Observable<ExplorePanelData> => {
-  if (data.tableFrames.length === 0) {
+  const tableFrames = data.tableFrames.length ? data.tableFrames : data.rawPrometheusFrames;
```

Comment:
Add comment as to why we're doing the same decoration with different frame types